query/src/main/java/org/infinispan/query/dsl/embedded/impl/JPAFilterAndConverter.java

```diff
@@ -116,14 +126,34 @@ public String toString() {
       @Override
       public void writeObject(ObjectOutput output, JPAFilterAndConverter filterAndConverter) throws IOException {
          output.writeUTF(filterAndConverter.jpaQuery);
+         Map<String, Object> namedParameters = filterAndConverter.namedParameters;
+         if (namedParameters != null) {
+            output.writeInt(namedParameters.size());
+            for (Map.Entry<String, Object> e : namedParameters.entrySet()) {
+               output.writeUTF(e.getKey());
+               output.writeObject(e.getValue());
+            }
+         } else {
+            output.writeInt(-1);
+         }
```

Comment:
The as commented previously: can you write an unsigned int or  0 when no params present?